text/0000-native-cli.md

```diff
@@ -0,0 +1,136 @@
+- Feature Name: native_cli
+- Start Date: 2020-09-17
+- RFC PR: (leave this empty)
+- Neon Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes making it possible to use Neon with only standard CLI tools: `npm` for JavaScript workflows, and `cargo` for Rust workflows. This should lighten the cognitive burden of learning Neon, make it easier to learn, and generally feel like an even simpler end-to-end user experience.
+
+# Motivation
+[motivation]: #motivation
+
+Historically, we created the `neon` command-line tool as a wrapper around the `npm` and `cargo` command-line tools to eliminate repetitive manual steps in the project creation and build steps. Excitingly, it appears there are enough extensibility points in these tools to be able to allow Neon users to use the native tooling for the usual workflows. This should have the following benefits:
+
+- Learnability: New Neon users can use the `npm` commands they're already used to, without having to learn another CLI tool.
+- Adoption: Lowering the barrier to entry should decrease a deterrence from adoption of Neon.
+- Maintainability: Neon users no longer need to worry about maintaining the right version of `neon` installed globally, and can just rely on the `npm` and `cargo` tools they already have installed on their system.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Creating a new Neon project
+
+Neon provides templates for conveniently creating various types of Neon projects, including:
+
+- libraries;
+- Node apps; and
+- Electron apps.
+
+### Libraries
+
+To create a new Neon library, use the command:
+
+```
+npm init neon-lib my-lib
+```
+
+Use whatever name you want for your library in place of `my-lib`.
+
+### Node apps
+
+To create a new Node app, use the command:
+
+```
+npm init neon-app my-app
+```
+
+Use whatever name you want for your library in place of `my-app`.
+
+### Electron apps
```

Comment:
It might be interesting to explain why electron apps are treated differently from normal apps.

It would be possible to move `electron-build-env` functionality into `build.rs` and automate the discovery of an `electron` dependency. I would not recommend this because of the complexity and potential for false positives, but it might be useful to explain why we did not choose this.

It's also possible that N-API combined with dynamic library loading, we might not need any special code to handle Electron.

Reply:
Can you help me understand this part? I don't quite understand how dynamic library loading could accomplish that. And if we can't, it seems like we don't yet fully have a story for how to deprecate `neon build`?